webapp/client/src/components/FormFieldConsentBox.js

```diff
@@ -0,0 +1,96 @@
+import PropTypes from "prop-types";
+import styled from "styled-components";
+import FormFieldScaffolding from "./FormFieldScaffolding";
+
+
+const ConsentBox = styled.div`
+  &.checkbox {
+    padding: 0;
+    margin-top: var(--spacing-02);
+    flex-wrap: inherit;
+  }
+
+  &.consent-box {
+    background-color: var(--bg-highlight-color);
+    padding: var(--spacing-04);
+  }
+
+  &.checkbox input {
+    width: 30px;
+    height: 30px;
+    opacity: 0;
+  }
+
+  &.checkbox input:focus + label {
+      box-shadow: 0 0 0 3px var(--focus-color);
+      background-color: var(--focus-color);
+  }
+
+  &.checkbox input:checked + label {
+      background-color: var(--link-color);
+      background-image: url("../images/checked.svg");
+      background-repeat: no-repeat;
+      background-size: 22px;
+      background-position: center;
+  }
+
+  & label.checkmark {
+    display: block;
+    width: 30px;
+    height: 30px;
+    cursor: pointer;
+    background: white;
+    position: absolute;
+    border: 2px solid var(--text-color);
+  }
+`;
+
+function FormFieldConsentBox({
+  fieldName,
+  fieldId,
+  value,
+  required,
+  autofocus,
+  labelText,
+  errors
+}) {
+  return (
+    <FormFieldScaffolding
+      {...{
+        fieldName,
+        errors,
+        cols: "12",
+      }}
+      hideLabel
+      render={() => (
+        <ConsentBox className="form-row checkbox consent-box col-lg-10">
+          <input type="checkbox" id={fieldId} defaultValue={value} required={required} autoFocus={autofocus}/>
+          {/* TODO: there should be only one label for an input */}
+          {/* eslint-disable-next-line */}
+          <label htmlFor={fieldId} className="checkmark" />
+          <label htmlFor={fieldId} className="col-sm-10 col-form-label ml-3 pt-0">
+            {labelText}
+          </label>
+        </ConsentBox>
+      )}
+    />
+  );
+}
+
+FormFieldConsentBox.propTypes = {
+  fieldName: PropTypes.string.isRequired,
+  fieldId: PropTypes.string.isRequired,
+  labelText: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
```

Comment:
Looks like it's not easy to require a specific component _name_, which is a real shame. That way we could say: either a string or a `<Trans>`.

The only way I see is this (untested), which is cumbersome:

```suggestion
  labelText: PropTypes.oneOfType([
    PropTypes.string,
    function(props, propName, componentName) {
      if (componentName !== 'Trans') {
        return new Error(`${propName}` must be a simple string or <Trans> component.`);
      }
    },
  ]).isRequired,
```

I think I'd prefer going with what you had.